scaleway/resource_baremetal_server_beta.go

```diff
@@ -0,0 +1,220 @@
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the server.",
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "ID of the server type.",
+				ValidateFunc: validationUUID(),
+			},
+			"image_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "The base image of the server", // TODO: add in doc example with UUID
+				ValidateFunc: validationUUID(),
+			},
+			"ssh_key_ids": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validationUUID(),
+				},
+				Required:    true,
+				ForceNew:    true,
+				Description: "Array of SSH key IDs allowed to SSH to the server.",
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 255),
+				Description:  "Some description to associate to the server, max 255 characters.",
+			},
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "Array of tags to associate with the server.",
+			},
+			"zone":            zoneSchema(),
+			"organization_id": organizationIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayBaremetalServerCreate(d *schema.ResourceData, m interface{}) error {
+	baremetalApi, zone, err := getBaremetalAPIWithZone(d, m)
+	if err != nil {
+		return err
+	}
+
+	name, ok := d.GetOk("name")
+	if !ok {
+		name = getRandomName("bm")
+	}
+	createReq := &baremetal.CreateServerRequest{
+		Zone:           zone,
+		Name:           name.(string),
+		OrganizationID: d.Get("project_id").(string),
+		Description:    d.Get("description").(string),
+		OfferID:        d.Get("type").(string),
+	}
+	if raw, ok := d.GetOk("tags"); ok {
+		for _, tag := range raw.([]interface{}) {
+			createReq.Tags = append(createReq.Tags, tag.(string))
+		}
+	}
+	res, err := baremetalApi.CreateServer(createReq)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(newZonedId(zone, res.ID))
+
+	_, err = baremetalApi.WaitForServer(&baremetal.WaitForServerRequest{
+		Zone:     zone,
+		ServerID: res.ID,
+		Timeout:  BaremetalServerWaitForTimeout,
+	})
+	if err != nil {
+		return err
+	}
+
+	installReq := &baremetal.InstallServerRequest{
+		Zone:     zone,
+		ServerID: res.ID,
+		OsID:     d.Get("image_id").(string),
+		Hostname: res.Name,
+	}
+
+	if raw, ok := d.GetOk("ssh_key_ids"); ok {
+		for _, sshKeyID := range raw.([]interface{}) {
+			installReq.SSHKeyIds = append(installReq.SSHKeyIds, sshKeyID.(string))
+		}
+	}
+
+	_, err = baremetalApi.InstallServer(installReq)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Should we wait for server to be installed ?

Reply:
This is why I moved the wait in the read, but I'll move it here